### PR TITLE
Add documentation for stream component

### PR DESCRIPTION
**Pull request by hunterjm**
**Description:**
Documents the stream component and the new camera service.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#21473

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Review by klaasnicolaas — DISMISSED** [review]

**Comment by dshokouhi** [comment]
@hunterjm I had just started testing this, currently looking to see if I have a supported camera.  

On my dev instance of HA I have things running on a raspberry pi. When attempting to start HA with `stream:` in my configuration I got the following error:

```
2019-03-12 08:49:59 ERROR (SyncWorker_5) [homeassistant.util.package] Unable to install package av==6.1.2: Command "/home/pi/home-assistant/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-udfl2b3t/av/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-ftn5zmh2/install-record.txt --single-version-externally-managed --compile --install-headers /home/pi/home-assistant/include/site/python3.6/av" failed with error code 1 in /tmp/pip-install-udfl2b3t/av/
2019-03-12 08:49:59 ERROR (MainThread) [homeassistant.requirements] Not initializing stream because could not install requirement av==6.1.2
2019-03-12 08:49:59 ERROR (MainThread) [homeassistant.setup] Setup failed for stream: Could not install all requirements.
```

After a little bit of googling the following commands allowed me to get past this error, so I think a note for manual install users would be helpful here:

```
sudo apt-get install -y python-dev pkg-config

sudo apt-get install -y libavformat-dev libavcodec-dev libavdevice-dev libavutil-dev libswscale-dev libavresample-dev libavfilter-dev
```

If you think its out of scope for this PR I will submit a change but I think it may be helpful to some users to get in the first release :)

**Comment by hunterjm** [comment]
@dshokouhi - The pyAV package is also available pre-built on https://www.piwheels.org/, which Hassbian has as a default package source, I believe.  Maybe we can add a note to add that?  Can you try removing those packages and installing from piwheels?

**Comment by dshokouhi** [comment]
@hunterjm it looks like wheels is not supported on python 3.6?  I have Python 3.6.7 on my raspberry pi.  I already had the default package source setup on my pi as I had just checked it from the link but it did not grab the packages.

```
Wheels provided support all Raspberry Pi models (Pi 3, Pi 2, Pi 1 and Pi Zero). Wheels are provided for Python 3.4 and 3.5. We plan to add support for Python 3.6, 3.7 and 2.7.
```

When I attempt remove the installed packages and try again I get the same error about it being unable to find the packages.  Below is part of the error when I try to manually install pyAV, when I grab the packages I linked above the install works fine.

```
    Could not find libavformat with pkg-config.
    Could not find libavcodec with pkg-config.
    Could not find libavdevice with pkg-config.
    Could not find libavutil with pkg-config.
    Could not find libavfilter with pkg-config.
    Could not find libswscale with pkg-config.
    Could not find libswresample with pkg-config.
```

I can provide the full error but don't want to spam this PR.

**Comment by hunterjm** [comment]
Ok. Good to know that piwheels doesn’t support 3.6 and 3.7.

I think we can add a warning message in the docs, but upon release it should work out of the box in new hassbian installs, HASS.io, and the official docker image.

**Comment by dshokouhi** [comment]
ok I will just add the note for manual installers who run into the error on how to fix it once this PR gets merged :)

File: source/_components/stream.markdown [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+title: "Stream"
+description: "Instructions on how to integrate live streams within Home Assistant."
+date: 2019-02-06 13:40
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: home-assistant.png
+ha_category: "Other"
+ha_release: 0.90
```
Comment (frenck) [inline source/_components/stream.markdown]:
```suggestion
ha_release: "0.90"
```

File: source/_components/stream.markdown [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+layout: page
+title: "Stream"
+description: "Instructions on how to integrate live streams within Home Assistant."
+date: 2019-02-06 13:40
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: home-assistant.png
+ha_category: "Other"
```
Comment (frenck) [inline source/_components/stream.markdown]:
```suggestion
ha_category: 
  - Other
```

File: source/_components/stream.markdown [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+ha_release: 0.90
+---
```
Comment (frenck) [inline source/_components/stream.markdown:16]:
```suggestion
ha_iot_class: Local Push
ha_qa_scale: internal
---
```

File: source/_components/camera.markdown [hunk 4]
```diff
@@ -72,6 +72,28 @@ action:
...
+
+| Service data attribute | Optional | Description |
+| ---------------------- | -------- | ----------- |
+| `entity_id`            |      no  | Name of entity to fetch stream from, e.g., `camera.living_room_camera`. |
+| `media_player`            |      no  | Name of media player to play stream on, e.g., `media_player.living_room_tv`. |
+| `format`            |      yes  | Stream format supported by `stream` component and selected `media_player`. (default: `hls`) |
+
+For example, the following action in an automation would send an `hls` live stream to your chromecast.
+
+{% raw %}
```
Comment (frenck) [inline source/_components/camera.markdown]: Raw should not be needed here.
```suggestion
```

File: source/_components/camera.markdown [hunk 5]
```diff
@@ -72,6 +72,28 @@ action:
...
+```yaml
+action:
+  service: camera.play_stream
+  data:
+    entity_id: camera.yourcamera
+    media_player: media_player.chromecast
+```
+{% endraw %}
```
Comment (frenck) [inline source/_components/camera.markdown]:
```suggestion
```

File: source/_components/camera.markdown [hunk 6]
```diff
@@ -72,6 +72,28 @@ action:
 ```
 {% endraw %}
 
+#### {% linkable_title Service `play_stream` %}
+
+Play a live stream from a camera to selected media player(s).  Requires `stream` component to be set up.
```
Comment (frenck) [inline source/_components/camera.markdown]: Additional space.
```suggestion
Play a live stream from a camera to selected media player(s). Requires `stream` component to be set up.
```

File: source/_components/stream.markdown [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+The Stream component provides a way to proxy live streams through Home Assistant.  The component currently only supports the HLS format.
```
Comment (frenck) [inline source/_components/stream.markdown]: A space to many...
```suggestion
The Stream component provides a way to proxy live streams through Home Assistant. The component currently only supports the HLS format.
```

**Review by frenck — CHANGES_REQUESTED** [review]

**Review by frenck — DISMISSED** [review]

**Review by fabaff — APPROVED** [review]

File: source/_components/stream.markdown [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ha_release: "0.90"
+ha_iot_class: Local Push
+ha_qa_scale: internal
+---
+
+The `stream` component provides a way to proxy live streams through Home Assistant. The component currently only supports the HLS format.
+
+## {% linkable_title Configuration %}
+
+To enable the random binary sensor, add the following lines to your `configuration.yaml` file:
```
Comment (dshokouhi) [inline source/_components/stream.markdown:22]: we have a random binary sensor here? :)

**Review by dshokouhi — COMMENTED** [review]
probably want to update this part

File: source/_components/stream.markdown (hunk 8 above)
Comment (fabaff) [reply source/_components/stream.markdown:22]: Thanks

**Pull request merged**